csharp-intermediate-topics/WhenToUseReaderWriterLockSlimOverLockInCSharp/WhenToUseReaderWriterLockSlimOverLockInCSharp/BaseReaderWriter.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Diagnostics;
+
+namespace WhenToUseReaderWriterLockSlimOverLockInCSharp
+{
+    public abstract class BaseReaderWriter
+    {
+        private readonly Stopwatch _stopwatch = new();
+        protected readonly List<int> _listOfNumbers = new();
+
+        public List<int> ListOfNumbers
+        {
+            get { return _listOfNumbers; }
+        }
+
+        protected int GetNextRandom()
+        {
+            var maxNumber = _listOfNumbers.Count > 0 ? _listOfNumbers.Count - 1 : 0;
+            return Random.Shared.Next(0, maxNumber);
+        }
```

Comment:
We can modify this further and actually access the list and pull a random item from it, then that code will also only be in one place. We can return `0` in the case of an empty list as well. Also, no need for the `-1` as the upper bound is exclusive in `Random.Next`